src/Snak/SnakObject.php

```diff
@@ -113,7 +113,24 @@ public function serialize() {
 	 * @param string $serialized
 	 */
 	public function unserialize( $serialized ) {
-		$this->propertyId = PropertyId::newFromNumber( unserialize( $serialized ) );
+		$this->propertyId = self::newPropertyId( unserialize( $serialized ) );
+	}
+
+	/**
+	 * @param mixed $unserialized
+	 *
+	 * @return PropertyId
+	 */
+	protected static function newPropertyId( $unserialized ) {
+		if ( is_int( $unserialized ) ) {
+			return PropertyId::newFromNumber( $unserialized );
+		} elseif ( is_string( $unserialized ) ) {
+			return new PropertyId( $unserialized );
+		} elseif ( $unserialized instanceof PropertyId ) {
```

Comment:
Probably none atm, it's here for completeness. I don't have strong feelings about having it.